client/my-sites/upgrades/domain-management/domain-connect/domain-connect-authorize.jsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import Main from 'components/main';
+import Notice from 'components/notice';
+import DomainConnectAuthorizeDescription from './domain-connect-authorize-description';
+import DomainConnectAuthorizeRecords from './domain-connect-authorize-records';
+import DomainConnectAuthorizeFooter from './domain-connect-authorize-footer';
+import { actionType, noticeType } from './constants';
+import wp from 'lib/wp';
+
+const wpcom = wp.undocumented();
+
+class DomainConnectAuthorize extends Component {
+	static propTypes = {
+		providerId: PropTypes.string.isRequired,
+		serviceId: PropTypes.string.isRequired
+	};
+
+	state = {
+		action: null,
+		dnsTemplateRecordsRetrieved: false,
+		dnsTemplateError: false,
+		noticeType: null
+	};
+
+	componentDidMount() {
+		const { providerId, serviceId, params, translate } = this.props,
+			{ domain } = params;
+
+		wpcom.getDnsTemplateRecords( domain, providerId, serviceId, params )
+			.then( data => {
+				this.setState( {
+					action: actionType.READY_TO_SUBMIT,
+					dnsTemplateConflicts: data && data.conflicting_records,
+					dnsTemplateRecords: data && data.new_records
+				} );
+			}, error => {
+				const errorMessage = error.message ||
+					translate( 'We aren\'t able to set up the service with the information given. Please check ' +
+						'with your service provider to make sure they provided all the correct data.' );
+
+				this.setState( {
+					action: actionType.CLOSE,
+					noticeType: noticeType.ERROR,
+					noticeMessage: errorMessage,
+					dnsTemplateError: true
+				} );
+			} ).then( () => {
+				this.setState( {
+					dnsTemplateRecordsRetrieved: true
+				} );
+			} );
+	}
+
+	handleClickConfirm = () => {
+		const { providerId, serviceId, params, translate } = this.props,
+			{ domain } = params;
+
+		this.setState( {
+			action: actionType.SUBMITTING,
+			noticeType: null
+		} );
+
+		wpcom.applyDnsTemplateSyncFlow( domain, providerId, serviceId, params )
```

Comment:
The idea is to have a separate `actions` file and fire Redux actions for these. Since we're not currently using these anywhere else, it's fine like this, but just fyi.